src/Console/Command/CompareCommand.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Command;
+
+use PhpCsFixer\Config;
+use PhpCsFixer\ConfigInterface;
+use PhpCsFixer\Console\ConfigurationResolver;
+use PhpCsFixer\FixerFactory;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Helper\TableCell;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Adamo Aerendir Crespi <hello@aerendir.me>
+ */
+final class CompareCommand extends Command
+{
+    const COMMAND_NAME = 'compare';
+
+    /**
+     * @var ConfigInterface
+     */
+    private $defaultConfig;
+
+    /**
+     * @var FixerFactory
+     */
+    private $fixerFactory;
+
+    /**
+     * @param null|FixerFactory $fixerFactory
+     */
+    public function __construct(FixerFactory $fixerFactory = null)
+    {
+        parent::__construct();
+
+        if (null === $fixerFactory) {
+            $fixerFactory = new FixerFactory();
+            $fixerFactory->registerBuiltInFixers();
+        }
+
+        $this->fixerFactory = $fixerFactory;
+        $this->defaultConfig = new Config();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName(self::COMMAND_NAME)
+            ->setDefinition(
+                [
+                    new InputOption('config', '', InputOption::VALUE_REQUIRED, 'The path to a .php_cs file.'),
+                    new InputOption('show-risky', '', InputOption::VALUE_NONE, 'Shows also the riky fixers.'),
+                    new InputOption('hide-in-use', '', InputOption::VALUE_NONE, 'Hides all the rules currently configured (and so, used) to highlight only the ones not already in use.'),
+                    new InputOption('dump', '', InputOption::VALUE_NONE, 'Dumps the comparing result in a copy-and-pastable format ready for the .php_cs file.'),
```

Comment:
:-1: this looks wrong

code shall filter by sth, here, by `isUsed` and `isConfigured`, which will be general solution to filter what we want to display together with replacement for current `hide-in-use` and `show-risky` options.

then, code shall provide different output formats, like a table by default and json as an possibility

Reply:
> then, code shall provide different output formats, like a table by default and json as an possibility

This whole purpose of this command is to have an evidence of current PHP-CS-Fixer version and the user config. What would be the added value of dumping in json? I see no case where this command is related to anything outside PHP world, and I've never seen a `.php_cs` file written in json or with json-related data.

The table format is the default, the `dump` version is to enable a copy-pasteble code to be inserted in the user `.php_cs`

> code shall filter by sth

@keradus is right, the `dump` flag must mirror the output of the command executed without the flag

Reply:
what I mean is to pick up a format and output in it, not randomly dump php code at the end ;)

Reply:
Solved: the `--dump` flag dumps exactly what is shown in the table and uses `var_export` to generate a valid array that is copy-and-pastable.